build-tools/packages/build-cli/src/commands/generate/changeset.ts

```diff
@@ -0,0 +1,263 @@
+	static summary = `Generates a new changeset file.`;
+	static aliases: string[] = ["changeset:add"];
+	static enableJsonFlag = true;
+
+	static flags = {
+		branch: Flags.string({
+			char: "b",
+			description:
+				"The branch to compare against. This is used to populate the list of changed packages.",
+			default: DEFAULT_BRANCH,
+		}),
+		empty: Flags.boolean({
+			description: "Create an empty changeset file.",
+		}),
+		all: Flags.boolean({
+			description: `Include ALL packages, including examples.`,
+			default: false,
+		}),
+		uiMode: Flags.string({
+			description: `Controls the mode in which the interactive UI is displayed. The 'default' mode includes an autocomplete filter to narrow the list of packages. The 'simple' mode does not include the autocomplete filter, but has better UI that may display better in some terminal configurations. This flag is experimental and may change or be removed at any time.`,
+			default: "default",
+			options: ["default", "simple"],
+			helpGroup: "EXPERIMENTAL",
+		}),
+		...BaseCommand.flags,
+	};
+
+	static examples = [
+		{
+			description: "Create an empty changeset.",
+			command: "<%= config.bin %> <%= command.id %> --empty",
+		},
+		{
+			description: `Create a changeset interactively. Any packages edited relative to the '${DEFAULT_BRANCH}' branch will be selected by default.`,
+			command: "<%= config.bin %> <%= command.id %>",
+		},
+		{
+			description: `You can select packages relative to a different branch using --branch.`,
+			command: "<%= config.bin %> <%= command.id %> --branch next",
+		},
+		{
+			description: `By default example and private packages are excluded, but they can be included with --all.`,
+			command: "<%= config.bin %> <%= command.id %> --all",
+		},
+	];
+
+	public async run(): Promise<{
+		branch: string;
+		selectedPackages: string[];
+		changesetPath?: string;
+	}> {
+		const context = await this.getContext();
+		const { all, branch, empty, uiMode } = this.flags;
+
+		if (empty) {
+			// TODO: This only works for the root release group (client). Is that OK?
+			const emptyFile = await createChangesetFile(context.gitRepo.resolvedRoot, new Map());
+			// eslint-disable-next-line @typescript-eslint/no-shadow
+			const changesetPath = path.relative(context.gitRepo.resolvedRoot, emptyFile);
+			this.logHr();
+			this.log(`Created empty changeset: ${chalk.green(changesetPath)}`);
+			return {
+				branch,
+				selectedPackages: [],
+				changesetPath,
+			};
+		}
+
+		const repo = new Repository({ baseDir: context.gitRepo.resolvedRoot });
+		const remote = await repo.getRemote(context.originRemotePartialUrl);
+
+		if (remote === undefined) {
+			// Logs and exits
+			this.error(`Can't find a remote with ${context.originRemotePartialUrl}`, { exit: 1 });
+		}
+		this.verbose(`Remote is: ${remote}`);
+
+		const {
+			packages: changedPackages,
+			files: changedFiles,
+			releaseGroups: changedReleaseGroups,
+		} = await repo.getChangedSinceRef(branch, remote, context);
```

Comment:
The reference is always origin/remote repo?

Reply:
I added a comment to clarify. The build-tools always look for a remote that matches `microsoft/FluidFramework`. And yes, it always compares to the remote branch, to make sure that it's as up-to-date as possible. It has surprised me how infrequently some folks update their local main/next branch heads.

Reply:
This is odd, because it's supposed to do that automatically. I think the `--all` argument was getting ignored, but I
made a change that should fix that.